bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeChecker.java

```diff
@@ -2390,7 +2390,8 @@ private static boolean checkIsLikeArrayType(Object sourceValue, BArrayType targe
                     return targetNumericTypes.size() == 1;
                 }
 
-                if (isNumericType(targetTypeElementType) && targetTypeElementType.getTag() != TypeTags.BYTE_TAG) {
+                if (isNumericType(targetTypeElementType) && targetTypeElementType.getTag() != TypeTags.BYTE_TAG &&
```

Comment:
If we just check targetTypeElementType == FLOAT_TAG || targetTypeElementType == DECIMAL_TAG

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/30710/commits/688ab907b7244cd39e24c251849c329e7f7fbe24